### DISCUSSION TITLE
Is it possible to use a menu item as a dialog trigger?

### DISCUSSION THREAD

**Discussion by tuananhlai** [adobe/react-spectrum#5729] [Q&A] [answered]
Hi everyone, I was wondering, does React Aria support using a menu item as a dialog trigger? The documentation doesn't seem to mention this use case. For example...

```tsx
      <MenuTrigger>
        <Button aria-label="Menu">☰</Button>
        <Popover>
          <Menu onAction={alert}>
            <DialogTrigger>
              <MenuItem id="open">Open Dialog</MenuItem>
              {(close) => <Dialog>Hello</Dialog>}
            </DialogTrigger>
            <MenuItem id="second">Item 2</MenuItem>
          </Menu>
        </Popover>
      </MenuTrigger>
```

Any help would be much appreciated. Thank you 🙇

**Comment by reidbarber** [comment, marked as answer]
This API isn't currently supported due to the types of children Menu expects, but you can always do this in a controlled way. Here is the example from the React Aria homepage where the edit and delete items open up dialogs: https://github.com/adobe/react-spectrum/blob/ee51290b98fdc7cad31957ed2dd0a486eb0c1427/packages/dev/docs/pages/react-aria/home/ExampleApp.tsx#L246-L247

**Reply by devongovett** [reply]
It's also impossible because the menu would unmount the dialog trigger when it closes causing the modal to disappear immediately. Moving the dialog outside the menu is the correct approach.

**Reply by tuananhlai** [reply]
Got it! Thank you for your help.
